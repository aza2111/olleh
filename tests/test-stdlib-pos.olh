//tests standard library functions

print reverse("anagram");
print getLength("anagram");

readDict("example.txt");

print getWordScore("anagram");
print checkWord("anagram");
print checkWord("asdkf");
setLetterScore({'x' -> 4});
print letterScores.get('x');

String anagram1 = scramble("anagram");
if (checkWord(anagram1) || (anagram(anagram1) != Null)):
        print "scrambled!";
end;


Player player = fresh Player{letters -> fresh list(3)};
setLetters(player);
setLetterScore({'x' -> 1});
int total = 0;
foreach lett player.letters:
        total = total + letterScores.get(lett);
end;
print total; //should be 3
