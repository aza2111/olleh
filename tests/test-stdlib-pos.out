margana
7
1
true
true
4
scrambled!
3
